src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -54,7 +54,9 @@ public string InstanceId
             {
                 if (_instanceId == null)
                 {
-                    _instanceId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID") ?? Guid.NewGuid().ToString("N");
+                    _instanceId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
+                        ?? Environment.MachineName.GetHashCode().ToString("X").PadLeft(32, '0');
```

Comment:
When not running on Azure, this will now base the InstanceId generation on the machine name.